qprofiler/src/org/qcmg/qprofiler/bam/BamSummaryReport.java

```diff
@@ -55,802 +59,803 @@
+
+            // ISIZE  is done inside readGroupSummary.pParseRecord
+
+            // MRNM
+            final int mateRefNameIndex = record.getMateReferenceIndex();
+            if (mateRefNameIndex == -1) MRNMLengths.increment(mateRefNameMinusOne);
+            else MRNMLengths.increment(mateRefNameIndex);
+
+            // RNAME & POS
+            parseRNameAndPos(record.getReferenceName(), record.getAlignmentStart(), readGroup);    // Position value
+
+            if (record.getReadPairedFlag()) {
+                if (record.getFirstOfPairFlag()) {
+                    p1Lengths.increment(record.getReadBases().length);
+                } else if (record.getSecondOfPairFlag()) {
+                    p2Lengths.increment(record.getReadBases().length);
+                }
+
+                // coverage
+                if (includeCoverage) {
+                    parseCoverage(record);
+                }
+            }
+
+            // MAPQ (Mapping Quality)
+            final int mapQ = record.getMappingQuality();
+            mapQualityLengths.increment(mapQ);
+
+            if (includeMatrices) {
+                matrix = mapQMatrix.get(mapQ);
+                if (null == matrix) {
+                    MAPQMatrix newMatrix = new MAPQMatrix();
+                    matrix = mapQMatrix.putIfAbsent(mapQ, newMatrix);
+                    if (null == matrix)
+                        matrix = newMatrix;
+                }
+                matrix.addToMatrix(record.getReadLength(), MatrixType.LENGTH);
+            }
+
+            // only TAGS, FLAGS, and CIGARS are always summarised
+            // TAGS
+            parseTAGs(record, matrix, record.getReadBases(), record.getReadNegativeStrandFlag());
+
+            // CIGAR
+            parseCigar(record.getCigar());
+
+            // Flags
+            flagIntegerCount.increment(record.getFlags());
+        }
+    }
+
+    private void parseTAGs(final SAMRecord record, final MAPQMatrix matrix, final byte[] readBases,
+                           final boolean reverseStrand) {
+
+        // if the record has a CQ tag, then it will likely have a CS tag
+        // if a record does not have a CQ tag, then it will not have a CS tag
+        String value = (String) record.getAttribute(SAMTag.CQ);
+        if (null != value) {
+            // CQ
+            byte[] valueB = SAMUtils.fastqToPhred(value);
+            if (null != valueB) {
+                SummaryByCycleUtils.parseIntegerSummary(tagCQByCycle, valueB);
+                SummaryReportUtils.tallyQualScores(valueB, cqBadReadLineLengths);
+            }
+            // CS
+            value = (String) record.getAttribute(SAMTag.CS);
+            if (null != value) {
+                SummaryByCycleUtils.parseCharacterSummary(tagCSByCycle, value, null, 1);
+                SummaryReportUtils.tallyBadReadsAsString(value, csBadReadLineLengths);
+            }
+        }
+
+        parseStringAttribute(record.getAttribute("RG"), tagRGLineLengths);
```

Comment:
I understand that this `parse*` naming is consistent with the usage already established in this file but would you consider using instead  `aggregate*` or `accumulate*` for these two new methods.

Reply:
yep - good idea